addons/hr_attendance/models/hr_attendance.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import api, fields, models
+from openerp.exceptions import UserError
+
+
+class HrActionReason(models.Model):
+    _name = "hr.action.reason"
+    _description = "Action Reason"
+    name = fields.Char('Reason', required=True, help='Specifies the reason for Signing In/Signing Out.')
+    action_type = fields.Selection([('sign_in', 'Sign in'), ('sign_out', 'Sign out')], default='sign_in')
+
+class HrAttendance(models.Model):
+    _name = "hr.attendance"
+    _description = "Attendance"
+    _order = 'name desc'
+
+    def _employee_get(self):
+        employee = self.env['hr.employee'].search([('user_id', '=', self.env.uid)], limit=1)
+        return employee and employee.id or False
+
+    name = fields.Datetime('Date', required=True, select=1, default=lambda self: fields.datetime.now())
+    action = fields.Selection([('sign_in', 'Sign In'), ('sign_out', 'Sign Out'), ('action', 'Action')], required=True)
+    action_desc = fields.Many2one("hr.action.reason", "Action Reason", domain="[('action_type', '=', action)]", help='Specifies the reason for Signing In/Signing Out in case of extra hours.')
+    employee_id = fields.Many2one('hr.employee', "Employee", required=True, select=True, default=_employee_get)
+    department_id = fields.Many2one('hr.department', "Department", related="employee_id.department_id")
+    worked_hours = fields.Float(compute='_compute_worked_hours', store=True)
```

Comment:
Where is _compute_worked_hours defined(It seems you forgot to change the name of the method) ?